repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
     description = '''No description has been provided for the InplaceUpgrade workflow.'''
 
     class FactsCollectionPhase(Phase):
+        '''Operating System Version Detection, IP, Hostname, RPMs, Services,
+        Augeas.
+        '''
```

Comment:
Get info about the system - what is installed, settings, ... IOW, scan the system to get facts about it.